lib/dbadaptor.py

```diff
@@ -150,28 +156,23 @@ def update(self, conn, table_name, values, where, ph):
         if where is not None:
             sql = sql + 'WHERE ' + where
         sql_cmd = sql % (table_name, sets)
-        c.execute(sql_cmd, vals)
+        with closing(conn.cursor()) as c:
+            c.execute(sql_cmd, vals)
         conn.commit()
 
-    def delete(self, conn, table_name, where):
-        '''Update data in a table
-        @conn A connection of database
+    def remove(self, conn, table_name, where):
+        '''Remove rows based on specified conditions
         @table_name(str) The table name
         @where(str) Selection condition which is mandatory here!
         '''
-        c = conn.cursor()
         sql = 'DELETE FROM %s WHERE %s' % (table_name, where)
-        c.execute(sql)
+        with closing(conn.cursor()) as c:
+            c.execute(sql)
         conn.commit()
 
 
 class SQLDatabaseAdaptor(DatabaseAdaptor):
 
-    def __init__(self, mes_level=1, log_file=None):
-        DatabaseAdaptor.__init__(self, mes_level, log_file)
-        self.mes_level = mes_level
-        self.log_file = log_file
```

Comment:
These two lines are indeed redundant. 
But I would keep the `__init__` for expandability